src/kmod.rs

```diff
@@ -0,0 +1,108 @@
+//! Load and unload kernel modules.
+
+use libc;
+use std::ffi::CStr;
+use std::os::unix::io::AsRawFd;
+
+use errno::Errno;
+use Result;
+
+/// Loads an ELF image into kernel space, performs any necessary symbol relocations,
+/// initializes module parameters to values provided by the caller, and then runs the module's init function.
+/// This function requires privilege.
+///
+/// The module_image argument points to a buffer containing the binary image to be loaded.
+/// The module image should be a valid ELF image, built for the running kernel.
+///
+/// The param_values argument is a string containing space-delimited specifications of the values for module parameters.
+/// Each of the parameter specifications has the form:
```

Comment:
Could use provide an example that actually passes parameters? This format specifier is technically correct but can be hard to understand. Providing an example would be helpful.